package/MDAnalysis/analysis/encore/covariance.py

```diff
@@ -0,0 +1,260 @@
+# covariance.py --- Covariance matrix calculations
+# Copyright (C) 2014 Wouter Boomsma, Matteo Tiberti
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Covariance calculation --- :mod:`encore.covariance`
+=====================================================================
+
+The module contains functions to estimate the covariance matrix of
+an ensemble of structures.
+
+:Author: Matteo Tiberti, Wouter Boomsma, Tone Bengtsen
+:Year: 2015--2016
+:Copyright: GNU Public License v3
+:Mantainer: Matteo Tiberti <matteo.tiberti@gmail.com>, mtiberti on github
+
+.. versionadded:: 0.14.0
+"""
+
+import numpy
+
+class EstimatorML(object):
+    """
+    Standard maximum likelihood estimator of the covariance matrix.
+    The generated object acts as a functor.
+    """
+    @staticmethod
+    def calculate(coordinates, reference_coordinates=None):
+        """
+        Parameters
+        ----------
+
+        coordinates : numpy.array
+            Flattened array of coordiantes
+
+        reference_coordinates : numpy.array
+            Optional reference to use instead of mean
+
+        Returns
+        -------
+
+        cov_mat : numpy.array
+            Estimate of  covariance matrix
+
+        """
+
+        if reference_coordinates is not None:
+
+            # Offset from reference (for a normal covariance calculation
+            # this would be the distance to the average)
+            coordinates_offset = coordinates - reference_coordinates
+
+            # Calculate covariance manually
+            coordinates_cov = numpy.zeros((coordinates.shape[1],
+                                           coordinates.shape[1]))
+            for frame in coordinates_offset:
+                coordinates_cov += numpy.outer(frame, frame)
+            coordinates_cov /= coordinates.shape[0]
+
+            return coordinates_cov
+
+        else:
+            return numpy.cov(coordinates, rowvar=0)
+
+    __call__ = calculate
```

Comment:
Is there a reason you are explicitly disguising functions as objects? Seems to be this could always be a default argument.